### DISCUSSION TITLE
How to use the built-in component "Image" from "next/image" in a production environment?

### DISCUSSION THREAD

**Discussion by Yukixieyuya821** [saltyshiomix/nextron#373] [Show and tell]
Hey there, folks!!!

Recently, I've encountered an issue with using the built-in component "Image" from "next/image" in a production environment, where images are not loading correctly.

![image](https://github.com/saltyshiomix/nextron/assets/53206696/5636c675-73f4-4cef-965c-4e1961adb2ed)

The issue occurs after packaging and installing the source code. Once opened, images fail to load.

![image](https://github.com/saltyshiomix/nextron/assets/53206696/87fb2483-1c2e-4fcc-a943-9374841453e1)

The problem arises after running next export, as it generates static assets that cannot handle _next/image. This is because handling such requests requires a dependency on the Next.js server. To resolve this issue, you can redirect the URL to point to a Next.js server before sending the request. Of course, you must first deploy a Next.js server.

main/background.ts
![image](https://github.com/saltyshiomix/nextron/assets/53206696/5cf540b8-7509-438b-9ba9-e728b4d27962)

After repackaging and reinstalling.
![image](https://github.com/saltyshiomix/nextron/assets/53206696/f5c10a60-7bdc-44a8-abd8-3c42fcd22555)

The images can now be loaded correctly on the page.

**Comment by bm777** [comment]
Hi @Yukixieyuya821 , I'm not familiar with .ts, can you provide an equivalent for background.js?

**Reply by Yukixieyuya821** [reply]
This code is consistent with js syntax

**Comment by bm777** [comment]
Hi @Yukixieyuya821  there is a normal workaround for this issue. Just adding domain in the next.config.js file like [here](https://github.com/saltyshiomix/nextron/issues/392#issuecomment-1693055028)

**Reply by Yukixieyuya821** [reply]
This solution is to solve the problem of images stored in third parties. If they are images from local projects, they cannot be mapped and require a next service.

**Reply by bm777** [reply]
right.
I think in my case, I was trying to access a file via another service (serving via fastapi)
